### DISCUSSION TITLE
Trigger a Webhook with a Webhook

### DISCUSSION THREAD

**Discussion by PaulR282** [netbox-community/netbox#13176] [Q&A]
Is there a way to trigger a Webhook when another Webhook is executed? I want to do 2 requests in a specific sequence. The first one is working great. If I select "Content types: Extras > Webhook", "Events: Job terminations" it won't trigger (I guess because Webhooks are not Jobs in netbox, although it is a job when you look in the django admin ui).

**Comment by candlerb** [comment]
I think you need an external workflow engine for this: Netbox webhook calls workflow engine, workflow engine calls other actions.

If you want a pretty framework then something like [Camunda 7](https://blog.bernd-ruecker.com/use-camunda-without-touching-java-and-get-an-easy-to-use-rest-based-orchestration-and-workflow-7bdf25ac198e) will work, if you can stomach a Java process running in your network; or there are python ones like [airflow](https://github.com/apache/airflow), [prefect](https://github.com/prefecthq/prefect) and [luigi](https://github.com/spotify/luigi). It's a while since I looked into options for these, but there's a big list [here](https://github.com/meirwah/awesome-workflow-engines).

Otherwise, you could make your first webhook call the second one just before it's finished.

**Reply by PaulR282** [reply]
Thanks for the ideas. How could I make the first one call another one within netbox? (last line)

**Reply by candlerb** [reply]
> Thanks for the ideas. How could I make the first one call another one within netbox? (last line)

What I meant was:

```
Netbox ----> Webhook receiver 1 ----> Webhook receiver 2
```

That is, you modify Webhook receiver 1 (i.e. the application which receives the webhook call) so that it calls Webhook receiver 2 directly, when it's finished its work.

Another option would be to write a small webhook proxy, which calls the other two hooks in sequence:

```
Netbox ----> Webhook proxy ----> Webhook receiver 1
                          `----> Webhook receiver 2
```

But arguably that's a custom-written, mini external workflow engine.  (Probably only requires a dozen lines of python though)

**Reply by PaulR282** [reply]
Ok, thanks. This would of course be an option but since it's not very dynamic, I'll stick with only calling webhooks with netbox and not calling the second one which is a bit optional.
